### PR TITLE
Add arguments to rand

### LINKED ISSUE #223: Numpy rand not working
rand from numpy.random cannot be pyccelized. This is marked as to do in tests/codegen/scripts/numpyext.py

### PR DESCRIPTION
**Pull request by EmilyBourne**
Fixes #223

### CODE REVIEW

File: pyccel/ast/numpyext.py [hunk 1]
```diff
@@ -881,30 +882,59 @@ def fprint(self, printer, lhs):
...
-    def __new__(cls, arg = Nil()):
-        return Real.__new__(cls, arg)
+    _dtype = 'real'
+    _precision = default_precision['real']
 
-    def __init__(self):
-        Real.__init__(self, None)
+    def __init__(self, *args):
+        if len(args) == 0:
+            self._shape = ()
```
Comment (yguclu) [inline pyccel/ast/numpyext.py:896]: Isn't `args` an empty tuple when its length is zero?

Comment (EmilyBourne) [reply pyccel/ast/numpyext.py]: You are right!

File: pyccel/ast/numpyext.py [hunk 2]
```diff
@@ -881,30 +882,59 @@ def fprint(self, printer, lhs):
...
+        stmts = []
+
+        if self.rank>0:
+            # Create statement for allocation
+            if not stack_array:
+                # Transpose indices because of Fortran column-major ordering
+                shape = self.shape[::-1]
+
+                if isinstance(self.shape, (PythonTuple,Tuple,tuple)):
+                    # this is a correction. problem on LRZ
```
Comment (yguclu) [inline pyccel/ast/numpyext.py:920]: What problem do we have on LRZ?

Comment (EmilyBourne) [reply pyccel/ast/numpyext.py:920]: I don't know. The allocation code was copied and pasted from the Full class (this should probably be done once. Maybe in the allocation super class that we could create for issue #307)
If I work my way back up the tree it seems that @saidctb wrote this comment originally

Comment (yguclu) [reply pyccel/ast/numpyext.py:920]: @saidctb Do you remember what this problem was about? Can we just remove those lines?

Comment (saidctb) [reply pyccel/ast/numpyext.py:920]: @ratnania added this part so I'm not sure

Comment (yguclu) [reply pyccel/ast/numpyext.py:920]: After taking another look at it, I think that we never enter the code at line 927 just below. In fact `shape` should never be a scalar, and the comment `# this is a correction. problem on LRZ` is probably a leftover. So I suggest we replace the whole `if/else` block with just
```
shape_code = ', '.join('0:' + printer(Add(i, Integer(-1))) for i in shape)
```
The same change should be made to the `Full` class, and maybe to other places too.

Comment (EmilyBourne) [reply pyccel/ast/numpyext.py:920]: I think you are right. I tried to ensure that shape is always a tuple, so it shouldn't be used. And codacy's coverage seems to indicate that it isn't used

**Pull request merged**